### PR TITLE
AArch64: Implement Vector Population Count evaluator

### PR DESCRIPTION
**Pull request by Akira1Saitoh**
This commit implements vpopcnt evaluator on AArch64.

### CODE REVIEW

File: compiler/aarch64/codegen/OMRTreeEvaluator.hpp [hunk 1]
```diff
@@ -438,6 +438,29 @@ class OMR_EXTENSIBLE TreeEvaluator: public OMR::TreeEvaluator
     */
    static TR::Register *inlineVectorMaskedBinaryOp(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op, binaryEvaluatorHelper evaluatorHelper = NULL);
 
+   typedef TR::Register *(*unaryEvaluatorHelper)(TR::Node *node, TR::Register *resReg, TR::Register *srcReg, TR::CodeGenerator *cg);
+   /**
+    * @brief Helper function for generating instruction sequence for unary binary operations
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRTreeEvaluator.hpp:443]: `unary binary` -> `unary`

File: compiler/aarch64/codegen/UnaryEvaluator.cpp [hunk 2]
```diff
@@ -116,14 +116,22 @@ TR::Register *OMR::ARM64::TreeEvaluator::lnegEvaluator(TR::Node *node, TR::CodeG
    return tempReg;
    }
 
-static TR::Register *inlineVectorUnaryOp(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op)
+TR::Register *OMR::ARM64::TreeEvaluator::inlineVectorUnaryOp(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op, unaryEvaluatorHelper evaluatorHelper)
    {
    TR::Node *firstChild = node->getFirstChild();
    TR::Register *srcReg = cg->evaluate(firstChild);
-   TR::Register *resReg = (firstChild->getReferenceCount() == 1) ? srcReg : cg->allocateRegister(TR_VRF);
```
Comment (knn-k) [inline compiler/aarch64/codegen/UnaryEvaluator.cpp:123]: Any reason for changing this line?

File: compiler/aarch64/codegen/OMRTreeEvaluator.hpp (hunk 1 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRTreeEvaluator.hpp]: Removed `binary`.

File: compiler/aarch64/codegen/OMRTreeEvaluator.hpp [hunk 3]
```diff
@@ -438,6 +438,29 @@ class OMR_EXTENSIBLE TreeEvaluator: public OMR::TreeEvaluator
...
+    * @param[in] node: node
+    * @param[in] cg: CodeGenerator
+    * @param[in] op: unary opcode
+    * @param[in] evaluatorHelper: optional pointer to helper function which generates instruction stream for operation
+    * @return vector register containing the result
+    */
+   static TR::Register *inlineVectorUnaryOp(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op, unaryEvaluatorHelper evaluatorHelper = NULL);
+
+   /**
+    * @brief Helper function for generating instruction sequence for masked unary binary operations
```
[the remark this answers is not in the document]
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRTreeEvaluator.hpp]: Removed `binary`.

**Comment by knn-k** [comment]
Jenkins build aarch64,amac

**Review by knn-k — APPROVED** [review]
See Issue #6516 for the socket test failures in porttest on macOS.

**Pull request merged**